Sources/Error Handling/PurchasesError.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchasesError.swift
+//
+//  Created by Nacho Soto on 8/31/22.
+
+import Foundation
+
+/// An error returned by a `RevenueCat` public API.
+public typealias PublicError = NSError
```

Comment:
Hmm I wonder if we should rename this... `PublicError` from the point of view of SDK users doesn't make that much sense... Maybe `RCError`? `PurchasesError` is already used... Can't think of other names at the moment 😅

Reply:
Hmm I guess I don't love any of the current options but I have a slight preference for `ExposedError`. Also, not sure if we should namespace the errors to RC? As in `RCExposedError`?

Reply:
We don't need to namespace in Swift (since it's already namespaces by the module name `RevenueCat`. And this `typealias` is transparent in Objective-C, meaning APIs see `NSError` instead.